qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -295,10 +296,15 @@ def _format_data(
                     continue
                 if len(g_values) == 1:
                     averaged[k] = v[0]
+                elif all(isinstance(v, numbers.Number) for v in g_values):
+                    averaged[k] = np.average(g_values)
                 else:
-                    unique = set(v)
+                    try:
```

Comment:
Sometimes there are metadata elements, such as `count_ops` in RB, that make the ScatterTable rather large if they're included. I wonder if we should consider not including this kind of metadata in the ScatterTable when they're used in the analysis but not needed for the fit itself, what do you think?